go-controller/pkg/util/ovs.go

```diff
@@ -125,6 +125,67 @@ type execHelper struct {
 
 var runner *execHelper
 
+type ExecUtilRunSvc interface {
+	RunCmd(cmd kexec.Cmd, cmdPath string, envVars []string, args ...string) (string, string, error)
+	Run(cmdPath string, args ...string) (string, string, error)
+	RunWithEnvVars(cmdPath string, envVars []string, args ...string) (string, string, error)
+}
+
+// ExecUtilRunSvcImplStruct implements the methods defined in the ExecUtilRunSvc interface
+type ExecUtilRunSvcImplStruct struct {
+}
+
+// RunCmd invokes the methods of the Cmd interfaces defined in k8s.io/utils/exec to execute commands
+// Note: the cmdPath and args parameter are used only for logging and is not processed
+func (runsvc *ExecUtilRunSvcImplStruct) RunCmd(cmd kexec.Cmd, cmdPath string, envVars []string, args ...string) (string, string, error) {
+	if cmd == nil {
+		return "", "", fmt.Errorf("cmd object cannot be nil")
+	}
+	if len(envVars) != 0 {
+		klog.V(5).Info("setting cmdSetEnv")
+		cmd.SetEnv(envVars)
+	}
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+
+	cmd.SetStdout(stdout)
+	cmd.SetStderr(stderr)
+
+	counter := atomic.AddUint64(&runCounter, 1)
+	logCmd := fmt.Sprintf("%s %s", cmdPath, strings.Join(args, " "))
+	klog.V(5).Infof("exec(%d): %s", counter, logCmd)
+
+	err := cmd.Run()
+	klog.V(5).Infof("exec(%d): stdout: %q", counter, stdout)
+	klog.V(5).Infof("exec(%d): stderr: %q", counter, stderr)
+	if err != nil {
+		klog.V(5).Infof("exec(%d): err: %v", counter, err)
+	}
+	return strings.Trim(strings.TrimFunc(stdout.String(), unicode.IsSpace), "\" \n"),
+		stderr.String(), err
+}
+
+func (runsvc *ExecUtilRunSvcImplStruct) Run(cmdPath string, args ...string) (string, string, error) {
+	cmd := runner.exec.Command(cmdPath, args...)
+	stdout, stderr, err := runsvc.RunCmd(cmd, cmdPath, []string{}, args...)
+	return stdout, stderr, err
+}
+
+func (runsvc *ExecUtilRunSvcImplStruct) RunWithEnvVars(cmdPath string, envVars []string, args ...string) (string, string, error) {
+	cmd := runner.exec.Command(cmdPath, args...)
+	stdout, stderr, err := runsvc.RunCmd(cmd, cmdPath, envVars, args...)
+	return stdout, stderr, err
+}
+
+var RunCmdExecSvcInst ExecUtilRunSvc
+
+func GetExecUtilRunSvc() ExecUtilRunSvc {
+	if RunCmdExecSvcInst == nil {
+		return &ExecUtilRunSvcImplStruct{}
+	}
+	return RunCmdExecSvcInst
+}
+
```

Comment:
Should GetExecUtilRunSvc, ExecUtilRunSvc, and RunCmdExecSvcInst all be exported? It is being used within the same util package, so should we still export it?

Reply:
At this time, I see that it would need to be exported so that Unit Test for functions in other packages can make use of it as well. For e.g; refer the line in file https://github.com/ovn-org/ovn-kubernetes/blob/1e2c6f54a1f23b23c11acd7bef615fd7ee78ba42/go-controller/pkg/ovn/pods_unit_test.go#L190 as part of upcoming (still WIP) PR at https://github.com/ovn-org/ovn-kubernetes/pull/1355

Reply:
@girishmg I have removed the GetExecUtilRunSvc method per your suggestion and `RunCmdExecSvcInst` is now renamed as `defaultExecRunner`. The interface definition `ExecRunner` remains exported so as to be able to use the mocks.